spec/rails_integration_spec.rb

```diff
@@ -106,14 +106,13 @@ def prepare_params(params)
   end
 
   describe "nested_array" do
-    it "responds with a 400 when the nested array is not supplied properly" do
-      params = {
-        'filter' => 'state'
-      }
+    it "responds with a 200 when the nested array is provided as nil" do
+      expect { get :nested_array, **prepare_params({ filter: { state: nil } }) }.not_to raise_error
+    end
 
-      expect { get :nested_array, **prepare_params(params) }.to raise_error do |error|
-        expect(error).to be_a(RailsParam::InvalidParameterError)
-      end
+    it "responds with a 400 when the nested array is passed as string" do
+      expect { get :nested_array, **prepare_params({ filter: { state: "a string" } }) }
```

Comment:
this will pass the following to recurrent calls to `ParamEvaluator`:

```
#<ActionController::Parameters {"filter"=>{"state"=>"a string"}, "controller"=>"fake", "action"=>"nested_array"} permitted: false>
#<ActionController::Parameters {"state"=>"a string"} permitted: false>
{0=>"a string"}
```

Reply:
My understanding is that this test is expected to return without raising the exception as everything seems valid. Have you added it just for illustration purpose?

Reply:
Both of my tests are currently failing. They return 200.

Reply:
Ok but this one should return 200, the payload seems valid

Reply:
@iMacTia On that note. In what universe `state: "a string"` coercing to `["a string"]` did sound like a good idea? :-D